### DISCUSSION TITLE
Markdown Reader Support

### DISCUSSION THREAD

**Discussion by Heroor** [Heroor/md-reader#64] [General]
# 💡 Need Help? We’re Here to Support You!  

Hello, Markdown Reader users! 👋  

If you’re experiencing issues, have questions, or need assistance with **Markdown Reader**, you’ve come to the right place. Let’s make sure you have the best experience possible! 🌟  

---

## 📌 Common Topics We Can Help With:  
1. **Installation Problems**  
   - Trouble installing Markdown Reader in your browser?  

2. **Feature Questions**  
   - Not sure how to enable or configure specific plugins?  

3. **Bug Reports**  
   - Encountered unexpected behavior?  
   - Let us know what’s happening so we can fix it!  

4. **Customization Requests**  
   - Looking for specific settings?  

5. **General Questions**  
   - Anything else we can assist you with?  

---

## 🛠️ How to Get Support  
To help us resolve your issue quickly, please provide the following:  

1. **A clear description of the problem**:  
   - What were you trying to do?  
   - What went wrong?  

2. **Your setup**:  
   - Browser name and version.  
   - Operating system (Windows, macOS, Linux, etc.).  

3. **Steps to reproduce** (if applicable):  
   - How can we recreate the issue on our end?  

4. **Screenshots or error messages** (if available):  
   - These are super helpful for understanding the problem!  

---

## 📬 Ways to Contact Us  

Post your question or issue in this thread, and we’ll reply as soon as possible.   

---

Thank you for reaching out! We’re here to make Markdown Reader the best Markdown experience for you. 🙌  

Looking forward to helping you,  

**The Markdown Reader Team**

**Comment by Heroor** [comment]
# 💡 需要帮助？我们随时为您提供支持！  

亲爱的 Markdown Reader 用户，您好！👋  

如果您在使用 **Markdown Reader** 时遇到问题、有疑问或需要帮助，来这里看看！让我们一起确保您获得最佳的使用体验！🌟  

---

## 📌 我们可以提供帮助的常见话题：  
1. **安装问题**  
   - 在浏览器中安装 Markdown Reader 时遇到困难？  

2. **功能疑问**  
   - 不清楚如何启用或配置特定插件？  

3. **错误报告**  
   - 遇到意外的行为？  
   - 告诉我们发生了什么问题，以便我们修复它！  

4. **定制化需求**  
   - 想要特定的主题或设置？  

5. **其他疑问**  
   - 任何其他需要我们协助的事项？  

---

## 🛠️ 如何获得支持  
为了帮助我们快速解决您的问题，请提供以下信息：  

1. **清晰的问题描述**：  
   - 您在尝试做什么？  
   - 出现了什么问题？  

2. **您的设备信息**：  
   - 浏览器名称和版本。  
   - 操作系统（Windows、macOS、Linux 等）。  

3. **重现步骤**（如适用）：  
   - 如何在我们的设备上重现问题？  

4. **截图或错误信息**（如有）：  
   - 这些对理解问题非常有帮助！  

---

## 📬 联系我们的方式  

在本帖中发表您的问题或意见，我们会尽快回复。   

---

感谢您的反馈！我们将竭尽全力为您提供支持，让 Markdown Reader 成为您的最佳 Markdown 工具！🙌  

期待为您提供帮助
 
**Markdown Reader 团队**

**Comment by Heroor** [comment]
# Failed to Preview Local File

### Allowing File Access Permission

#### **For Chrome**  
1. Open Chrome's extensions management page: Go to `chrome://extensions/` in the address bar.  
2. Find the installed **Markdown Reader** extension and click on **Details**.  
3. Scroll down to find the **Allow access to file URLs** option.  
4. Enable this option.

> **Note**: Chrome, by default, restricts extension access to local files. You need to manually enable this permission in order to preview local markdown files. Markdown Reader only reads and renders markdown files and does not modify or upload user data. You can enable this option with confidence.

#### **For Firefox**  
By default, Firefox allows local file access. Ensure your file path starts with `file://`, for example: `file:///C:/path/to/your/file.md`.  
If you encounter issues, check for any potential restrictions, such as enhanced privacy settings, that might limit file access.

---

# 无法预览本地文件的解决办法

### 开启本地文件访问权限

#### **在 Chrome 中**  
1. 打开 Chrome 的扩展程序管理页面：在地址栏输入 `chrome://extensions/`。  
2. 找到已安装的 **Markdown Reader** 扩展程序并点击 **详细信息**。  
3. 向下滚动，找到 **允许访问文件网址** 选项。  
4. 启用此选项。

> **注意**：Chrome 默认禁止扩展访问本地文件，您需要手动启用此权限，以便正常预览本地 Markdown 文件。Markdown Reader 仅读取和渲染 Markdown 文件，不会修改或上传用户文件数据，可以放心开启。

#### **在 Firefox 中**  
默认情况下，Firefox 支持本地文件访问。确保您的文件路径以 `file://` 开头，例如：`file:///C:/path/to/your/file.md`。  
如果遇到问题，请检查是否有其他潜在的限制，例如增强的隐私设置，可能会限制文件访问。

**Comment by Heroor** [comment]
# Unable to Preview HTTP Links

### Check MIME Type Configuration

When trying to preview a markdown file from an online URL or local server, it's important to ensure that the correct MIME type is set for markdown files. If the MIME type is incorrectly set, the browser may treat the resource as an unknown file type and attempt to download it instead of displaying the content.

#### **Common Correct MIME Types**  
- `text/markdown`  
- `text/plain`

These MIME types allow the browser to recognize and display the markdown content properly. If the MIME type is set to `application/octet-stream` or another unknown/incorrect type, the browser may prompt for a download instead of rendering the file.

### **How to Fix This**  
- **For online resources**: If you're using an online server to host the markdown file, make sure the server is configured to send the correct MIME type in the response headers.  
  - For example, if using Apache, you can add the following line in your `.htaccess` file:  
    ```
    AddType text/markdown .md
    ```
- **For local server setup**: If you're hosting markdown files locally (for example, with a local web server), ensure that the server is configured to serve the correct MIME type for `.md` files.

> **Tip**: If the issue persists, you can check the file's response headers using the browser's developer tools (Network tab) to confirm the MIME type.

---

# 无法预览 HTTP 链接的解决办法

### 检查 MIME 类型设置

当尝试从在线 URL 或本地服务器预览 markdown 文件时，确保服务器返回正确的 MIME 类型非常重要。如果 MIME 类型设置不正确，浏览器可能会将该资源视为未知文件类型，并尝试下载它，而不是展示内容。

#### **常见的正确 MIME 类型**  
- `text/markdown`  
- `text/plain`

这些 MIME 类型让浏览器能够正确识别并展示 markdown 内容。如果 MIME 类型被设置为 `application/octet-stream` 或者浏览器未知的非法类型，浏览器可能会弹出下载提示，而不是渲染文件。

### **如何修复**  
- **在线资源**：如果您使用在线服务器托管 markdown 文件，请确保服务器设置了正确的 MIME 类型返回文件。  
  - 例如，若使用 Apache 服务器，可以在 `.htaccess` 文件中添加以下内容：  
    ```
    AddType text/markdown .md
    ```
- **本地服务器设置**：如果您在本地服务器上托管 markdown 文件（例如使用本地 Web 服务器），确保该服务器配置为正确提供 `.md` 文件的 MIME 类型。

> **提示**：如果问题仍然存在，可以使用浏览器的开发者工具（网络标签）检查文件的响应头，确认 MIME 类型。

**Comment by HHN0609** [comment]
好奇，大佬是怎么绕过CSP白名单机制，把脚本注入到web页面，然后把渲染好的MD注入进页面的

**Reply by Heroor** [reply]
脚本注入是扩展程序去执行的，所以不会有 CSP 问题，剩下的就是处理页面内容的逻辑了，没有很复杂，这个组织里保留了2.0版本的仓库，可以参考一下源码

**Reply by HHN0609** [reply]
![Clipboard_Screenshot_1741270039](https://github.com/user-attachments/assets/42ee60b2-2350-46b3-a271-af89c8c9128e)
我content-script里的文件会被拒绝😂。还有想借此问下，大佬是如何做到让页面不闪动的，还有如何判断https://github.com/mozilla/pdf.js/blob/master/README.md，这样的连接

**Reply by Heroor** [reply]
对了，你的 content-script 文件是多个 bundle 吗，如果是的话可以调整一下你构建的配置，content-script 只打成单个 js 文件，不要有异步 js 的加载；还有一种办法是使用 `chrome-extension://` 协议的绝对路径加载你扩展程序文件，应该不会遇到 CSP 限制，相对比较麻烦，可以尝试。

页面闪动我不太清楚你说的是哪种情况，可以在适当的时候用一些 loading 或者骨架屏之类的做过渡吧。

URL 的判断可以在 `manifest` 里设置给 `content_scripts.matches`，让插件只在你允许的 URL 下运行，这样可以省去很多判断；不然就手动匹配一下 URL，正则写对应该就行了。

**Reply by HHN0609** [reply]
thanks

**Comment by StrollStars** [comment]
请问可否支持网页渲染，比如一些在线记事本，用MD写，如果没有大纲栏标题的效果就打折扣。

**Reply by Heroor** [reply]
不太清楚在线记事本具体形式是怎样的，方便截个图或者留个链接嘛

**Reply by StrollStars** [reply]
比如[Rentry.co](https://rentry.org/)是原生支持MD的，还有[Note.ms](https://note.ms/)这种是纯记事本，这两种有可能都支持吗？

**Reply by Heroor** [reply]
感谢提供网站链接，我看了下 这类笔记网站展示的页面已经是渲染好的 HTML 了，而不是原始的 markdown 内容，这和 Markdown Reader 所适用的场景还是不太一样的，Markdown Reader 更适用于直接读取 markdown 原数据(RAW)页面，比如以 `.md` 为结尾的链接。

Rentry.co 好像支持以 `Rentry.co/raw/xxxx` 的方式获取到笔记的原数据，不过这个功能目前需要向 Rentry 官方申请开通，我现在无法看到这个链接会返回什么内容，如果是返回 markdown 原数据，那么我可以让 Markdown Reader 特殊支持 `Rentry.co/raw/*` 域名，且不限制 .md 扩展名，这样就可以让 Markdown Reader 渲染页面了。

Note.ms 我发现他自身支持在路径后添加 `.md` 后缀进入预览模式，但是同样是渲染后的 HTML，无法获取到 md 原数据，如果你知道他是否有什么 API 用来获取 raw 可以告诉我。

**Reply by StrollStars** [reply]
关于`Rentry.co`，你打算去申请开通吗？如果打算，可不可以从`rentry.org/raw/xxxx`读取，作用在`rentry.org/xxx`上？毕竟用户看的是后者。
PS：它的网站是`rentry.org`，`rentry.co`可能是早先用的。

关于`Note.ms`，感谢告知添加`.md`可以预览，它之前用的是[Minimalist Web Notepad](https://github.com/pereorga/minimalist-web-notepad)，但是该项目自带的演示网站无法通过添加`.md`预览，不知`Note.ms`是在原来基础上修改，还是另起炉灶，如果前者你可以试试这个项目。

HTML格式和MD可以互转，不过处理起来过于繁琐，而且可能出错吧？

**Reply by Heroor** [reply]
HTML 与 markdown 互转确实存在很多不确定性，无法准确还原原始 markdown 数据，且二次渲染的效果并不一定就有原来的好（尤其是这些笔记网站会有一些特殊的全局样式配置）。

Rentry 的 raw 服务貌似比较偏向笔记作者的个人定制化，公开暴露在浏览器扩展里抓取任意页面资源，在安全性和隐私性等方面并不是很合适的做法，而且从展示效果来看，Rentry 有一些特殊的全局样式的设置，展示其渲染内容最理想方式还是 Rentry 自己。

如果你只是需要获取到笔记的标题大纲，那么你倒是可以尝试专门为 Rentry.co 或 Note.ms 这类网站开发一个辅助扩展程序（毕竟我们是爱折腾的程序员对吧～），单独为他们提供标题解析功能才是最好的解决办法，如果你有兴趣或者需要帮助，我可以和你讨论一些设计思路。

再次感谢你分享这两个网站，我看到了很多有意思的"博客"或笔记，这个圈子非常有趣！

**Reply by StrollStars** [reply]
既然提到了HTML与markdown的互转，想顺带请教下，对于网页的保存，相比markdown，还是用HTML本身更合适对吧？

我个人也觉得，如果网站本身能实现是最好。我非程序员，不知道对于网站本身而言，这个实现起来是不是比较麻烦，因为目前接触到支持markdown格式的网站都没做。
不过现在借助AI，应该做起来比以前容易得多吧，目前打算用JavaScript脚本去实现。

感谢告知`Note.ms`添加`.md`的方式能用markdown去渲染，没提醒还真发现不了。

**Reply by Heroor** [reply]
markdown 其实只能算作是 HTML 的"子集"吧，或者说是 XML 的子集？毕竟 markdown 最后也是要被渲染成 HTML 页面的，HTML 的表现力终究还是比普通的 markdown 语法要强很多的，当然也要配合 CSS JavScript 等等。还有些 markdown 框架或库 甚至可以集成一些自定义前端组件，拉高了 markdown 的天花板。

然后你说的网页保存，是指你想把笔记类的内容保存成网页吗？如果是这样的话，写 markdown 还是现在比较主流的编写方式，比如现在的个人博客网站也是用一些 markdown 解析工具将 md 写的文章渲染成 HTML 博客页面的。（不过用 HTML 记笔记这事我真干过。。。）

开发一个解析大纲的浏览器扩展还是相对容易的，之前是不是程序员也没关系，就当这是你入门的一个机会吧～～

我有空也会再尝试下有没有别的方法为这两个网站做兼容，如果有好消息我会来告诉你！

**Reply by StrollStars** [reply]
是的，markdown的语法简洁或者说简单，几分钟就可以看完，纯HTML都比它复杂多了。至于框架和库甚至是自定义前端组件，它的兼容性就差了吧？或者只局限于当前平台？

我说的网页保存，就是字面意思，就是把网页保存到本地，方便离线查看以及防止链接失效。
我目前保存的是网页格式，但是有些浏览器扩展是把网页保存成markdown格式的。

你的意思是用扩展而不是脚本的形式实现更好是吗？我当初还想用CSS，但怎么想都觉得不可能。

感谢你的指教，我试试，有问题再来请教。
我目前写笔记是用的markdown语法，因为简洁高效，如果用HTML，怕会打断思路。

**Reply by Heroor** [reply]
你用的保存为 md 格式的工具，可能它是把网页的主要信息导出为 md 了吧，不知道转换效率高不高，这个看个人喜好吧。想离线查看原网页，我感觉其实就直接浏览器 ctrl+s 保存就最省事了。

回到解析大纲的问题，如果你想再简单点，可以直接写写`油猴脚本`，配合一些css应该就行了，AI 应该可以帮助你完成，加油💪！

**Reply by StrollStars** [reply]
浏览器`ctrl+s`保存，如果是单一文件只能保存成mht格式，这个格式已经过时了，所以一般都是通过第三方扩展，另一方面还能压缩。md除了效率，怕还会丢失网页的一些样式。

哈哈，那我就先试试油猴脚本，谢谢指教。

**Comment by philiprhoades** [comment]
Why am I still blocked?  I have signed up for the Pro trial but I still can't see my files from Brave browser using:

file:///home/phr/

This dir list is greyed out and the upgrade popup can't be closed.

UPDATE:  IT IS WORKING NOW - IT TOOK SOME TIME TO ALLOW ME ACCESS!

Phil.

**Comment by Rikuron** [comment]
Just curious if this extension can support rendering tables? I'm trying to type it like this but it's not rendering for me:


```
| Name | Age | Role |
|------|-----|------|
| Ken  | 21  | Dev  |
```

Did I do something wrong here?

**Reply by Heroor** [reply]
I just tried the rendering and it worked fine. The picture is the result of the rendering. What does your page look like?

<img width="517" height="160" alt="CleanShot 2025-09-08 at 17 51 30" src="https://github.com/user-attachments/assets/f47bad30-e5df-4a11-82b1-51b2c786ded2" />

**Reply by Rikuron** [reply]
It looks like this in mine. Did I configure something wrong maybe? I'm using it in Opera GX browser btw, could that be a factor?

<img width="397" height="200" alt="image" src="https://github.com/user-attachments/assets/fdfc2a22-213f-439e-a0ad-d5583c679549" />

**Reply by Heroor** [reply]
Are the other Markdown syntaxes rendered normally? Have you tried switching to another browser to see if it works properly? Maybe the Opera GX browser has compatibility issues because it hasn't been specifically adapted for this purpose.

It should also be noted that when previewing local files (file://xxx), you need to enable the extension's access permission for local files URL. Otherwise, what you will see is the source code of Markdown.

**Reply by Rikuron** [reply]
The others are rendering normally. I've also tried it with Chrome but the results are the same.

**Reply by Heroor** [reply]
This is extremely confusing. If it's just the absence of any rendering in the table, I've never encountered such a situation before. Is there any error message displayed in the browser's console?

**Reply by Rikuron** [reply]
No errors seen in my console

**Reply by Rikuron** [reply]
OH OKAY I FIGURED IT OUT! It was an error in the way I wrote the markdown file itself. Thank you for your replies and the concern, I appreciate it!

**Reply by Heroor** [reply]
If it can be solved, that would be great! I’ve started doubting myself, hahaha

**Comment by zhangfanpc** [comment]
需要升级到 mermaid 的最新版本11.10、最好是有一个自动升级的选项。当前是Syntax error in text    mermaid version 11.4.1

**Reply by Heroor** [reply]
收到，近期会尝试升级👌

**Comment by DRNadler** [comment]
We Need Tables!!!
Everything else is working fine, but not tables.
Thanks!
PS: Works if I add the header separation line! That's optional, right?
[tableNFG.md](https://github.com/user-attachments/files/23664876/tableNFG.md)

**Reply by Heroor** [reply]
The table can be rendered normally, but it is necessary to add the `|--|--|` divider line for it.

**Reply by DRNadler** [reply]
OK, thanks - I thought separator was optional...
Thanks for the nice work!

**Comment by Gitup78** [comment]
Hello,   
I signed up for pro recently, and I'm not able to sign in to access folder browsing on my work computer.
My work network has security restrictions that block access to your external authentication server, and I'm getting a 403 Forbidden error when attempting to sign in.
Is there a way to access the Pro features without having to sign in, like with a license file? 
- Chrome Version 143.0.7499.170. Windows 11

**Reply by Heroor** [reply]
I sincerely apologize. Pro features require permission verification and cannot be provided to guest users. If you are unable to use them, I recommend unsubscribing—cancellations within two days will not incur any charges. I am very sorry for the inconvenience.

**Comment by johannbol** [comment]
Hi!
I am trying to login but it seems that I have the wrong password. I am trying to change the password but clicking the "Continue" button does nothing.
<img width="430" height="324" alt="image" src="https://github.com/user-attachments/assets/057a419f-e20f-4aa5-89a3-329c6427173e" />
I am using Mac and have tried Safari, Vivaldi and Edge browsers but nothing works.

**Reply by Heroor** [reply]
Thank you for your feedback. The issue has been fixed. Please try again.

**Reply by johannbol** [reply]
Thanks, I am back :)

**Comment by daniel-rutten-businessfundamentals-nl** [comment]
Hi team,

I love the MD Reader for Edge.

Do you have an indication for a release date for the Safari extension?

- daniel

**Comment by Repoman999** [comment]
Please help me receive verification email to start an account. I have installed in Chrome version 149.0.7827.103 (Official Build) (arm64) on MacOS 26.5.1 (25F80). I try to sign up, but no veriication comes to my email inbox, junk, promo, or other folder. Your app was recommended by my IT department, so they do not bock this. How to proceed?

**Reply by Heroor** [reply]
Sorry for this issue. Please send the email address used for registration to `mkdreader@gmail.com`. I will help you check the problem.

**Reply by Repoman999** [reply]
Thank you, Heroor, it is dkeller@teamworkcommerce.com

**Comment by REmenecker-SR** [comment]
**Custom CSS Functionality Question**

Is it possible to use an `@import()` or other method to load custom CSS from an external local file, rather than pasting custom CSS into the settings field? 

I ask because I use this extension on both Chrome and Brave, and having a single location for custom CSS would make it easier to use across the browsers.

I have tried the following, but it did not work:
```
@import("file:///Users/myuserdir/Documents/misc/markdown-reader.css");
```

I currently have v3.6.24 of the free plugin running in Chrome and Brave.

**Reply by Heroor** [reply]
Currently, importing custom CSS files is not supported. This sounds like a great feature suggestion and I will try to implement it.

**Reply by REmenecker-SR** [reply]
> This sounds like a great feature suggestion and I will try to implement it.

Thank you for considering it, @Heroor!

**Reply by Heroor** [reply]
Hi! I tested this and confirmed that Markdown Reader already supports importing a local CSS file.

The issue is the @import syntax. Please use:

```
@import url("file:///Users/myuserdir/Documents/misc/markdown-reader.css");
```

Alternatively:

```
@import "file:///Users/myuserdir/Documents/misc/markdown-reader.css";
```

Place the @import statement at the beginning of the Custom CSS field, click Apply CSS, and refresh the Markdown page.

For local Markdown files, please also enable Allow access to file URLs on Markdown Reader’s extension details page in both Chrome and Brave.

You only need to add this import line once in each browser. Both browsers can then use the same CSS file, so future changes only need to be made in that file.

Note that importing a file:// CSS file works when viewing local `file://` Markdown documents. Browsers do not allow HTTP or HTTPS pages to load local `file://` stylesheets.

**Reply by REmenecker-SR** [reply]
> The issue is the **@import** syntax

Oh crap! Wow! Glad that's all it was, but embarrassed to suck up your time over a simple mistake like that.
